qiskit_ibm_runtime/utils/backend_decoder.py

```diff
@@ -28,16 +28,23 @@
 
 logger = logging.getLogger(__name__)
 
+FRACTIONAL_GATES = ("rzz", "rx")
+DYNAMIC_INSTRUCTIONS = ("if_else", "while_loop", "for_loop", "switch_case")
```

Comment:
Is it possible to check whether a gate is an instance of `ControlFlowOp`? I find it more robust than hard-coding instruction names.

Reply:
No, the mapping of gates to names is always done through hard-coding. For example [here](https://github.com/Qiskit/qiskit/blob/b9d5c9c6aeb4568b6e9fba8943517ebed300886d/qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py#L459-L464) in Qiskit SDK or in this section related to the filtering in `convert_to_target` in this repo:

https://github.com/Qiskit/qiskit-ibm-runtime/blob/da0a2d763d5592e8e3e472ecfcc17d7915b065b7/qiskit_ibm_runtime/utils/backend_converter.py#L79-L84

In Qiskit, there is a public function called `get_standard_gate_name_mapping` that gives a map from string names to quantum gates. It is still effectively [hard-coded](https://github.com/Qiskit/qiskit/blob/b9d5c9c6aeb4568b6e9fba8943517ebed300886d/qiskit/circuit/library/standard_gates/__init__.py#L85) but  at least you can reference it and count on it to be updated independently in the future. Perhaps something like that could be added for these other instructions as well.

We could stop hard-coding `FRACTIONAL_GATES` since I restored `is_fractional_gate`. I didn't mind the hard-coding of rx and rzz here though because `is_fractional_gate` also somewhat arbitrarily hard-codes exceptions to fractional gates and I don't know what the chances are that another gate will need to be supported but it only requires a small change:

https://github.com/Qiskit/qiskit-ibm-runtime/blob/da0a2d763d5592e8e3e472ecfcc17d7915b065b7/qiskit_ibm_runtime/utils/backend_converter.py#L354-L359

Reply:
I made all the code use the `is_fractional_gate` function. I didn't unify the cases of dynamic circuit instructions but I did discover that Qiskit 1.3 at least provides a variable with all the string names of control flow instructions so I used that for the filter (when it could be imported).